src/directives/offense/poisonRoom.ts

```diff
@@ -1,169 +1,188 @@
+			this.suspendColony(this.room.name);
+		}
+
+		if (Game.time % RUN_TIMER == 0 || this.room.constructionSites.length == 0) {
+			// capture variables
+			this.walkableSourcePosisions	 = _.filter(_.flatten(_.map(this.room.sources,
+													s => s.pos.neighbors)),pos => pos.isWalkable(true));
+			this.walkableControllerPosisions =  _.filter(this.room.controller!.pos.neighbors,
+													pos => pos.isWalkable(true));
+			
+			// reverse reservation if needed
+			if(this.room.controller.reservation && this.room.controller.reservation.ticksToEnd > 500) {
+				DirectiveControllerAttack.createIfNotPresent(this.room.controller.pos,'room');
+			}
+
+			// send fighters if needed
+			if(this.room && this.room.playerHostiles.length > 0) {
+				DirectiveOutpostDefense.createIfNotPresent(this.room.controller.pos,'room');
+			}
+		}
+	}
+	private suspendColony(roomName: string): string {
+		if (Overmind.colonies[roomName]) {
+			const colonyMemory = Memory.colonies[roomName] as ColonyMemory | undefined;
+			if (colonyMemory) {
+				colonyMemory.suspend = true;
+				Overmind.shouldBuild = true;
+				return `Colony ${roomName} suspended.`;
+			} else {
+				return `No colony memory for ${roomName}!`;
+			}
+		} else {
+			return `Colony ${roomName} is not a valid colony!`;
+		}
+	}
+
+	private poisonActions() {
+		if(!(this.room && this.room.controller && this.room.controller.level == 2 && 
+			 this.walkableControllerPosisions && this.walkableControllerPosisions)) {
+			return;
+		}
+
+		// Poison Controller First
+		if(this.walkableControllerPosisions.length > 0) {
+			_.first(this.walkableControllerPosisions).createConstructionSite(STRUCTURE_WALL);
+			return;
+		} 
+
+		// Poison Sources Next
+		// Trick: Csite are not walkable, wait for poisoner to carry energy before blocking Sources
+		// Trick: Remove sources Csites if poisoner need to harvest, i.e carrying 0 energy
+		if (this.overlords.roomPoisoner.roomPoisoners.length &&
+			this.overlords.roomPoisoner.roomPoisoners[0].carry.energy > 0) {
+				if(this.walkableSourcePosisions.length) {
+					_.first(this.walkableSourcePosisions).createConstructionSite(STRUCTURE_WALL);
+				}
+		} else {
+				_.forEach(this.room.constructionSites, csite => {csite.remove();} );
+		}
+		
+	}
+	private isPoisoned(): boolean {
+		let result = false;
+		if (this.room && this.room.controller!.level > 1) {
+			result = !!this.walkableSourcePosisions && !this.walkableSourcePosisions.length &&
+					 !!this.walkableControllerPosisions && !this.walkableControllerPosisions.length;
+			return result;
+		} else {
+			return false;
+		}
+	}
+
+	private prePoisonActions(): void {
+		if(!(this.room && this.room.controller && this.room.controller.my)) {
+			return;
+		}
+		// kill claimer if room claimed, it can blocking csite creation
+		// it can also be used as a quick energy source from the tomb
+		if (this.overlords.claim.claimers.length) {
+			this.overlords.claim.claimers[0].suicide();
+		}
```

Comment:
Mmm nice. Maybe kill all claimers just in case there is more, and remove the claim directive. This should probably be in it's own handleRoomClaiming method

Reply:
this should go on the overlord not the directive - never directly control creeps from the directive

Reply:
I am using an existing ClaimingOverlord so it is just one claimer.
should I modify it so the claimer suicide once the room is claimed?